lib/mv_opentelemetry/ecto.ex

```diff
@@ -75,18 +66,32 @@ defmodule MvOpentelemetry.Ecto do
     ]
 
     all_attributes = result ++ base_attributes ++ time_attributes(measurements)
-
     span_name = name(config, event, source)
-    span_opts = %{start_time: start_time, attributes: all_attributes}
+    span_opts = %{start_time: start_time, attributes: all_attributes, kind: :client}
+
+    span = Tracer.start_span(span_name, span_opts)
+
+    case query_result do
+      {:error, error} ->
```

Comment:
Why do you first match on an error case rather than successful one?

Reply:
Because the successful case doesn't matter - the value is not re-used elsewhere.

OpenTelemetry.Span.set_status/3 is a side-effectful function that manipulates data in ets.